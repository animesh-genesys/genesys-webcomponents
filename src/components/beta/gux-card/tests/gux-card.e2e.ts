import { newSparkE2EPage, a11yCheck } from 'test/e2eTestUtils';

describe('#render different card variations eg(outline,outline(explicit),filled,raised)', () => {
  [
    '<gux-card-beta><span>This is a content slot</span></gux-card-beta>',
    '<gux-card-beta accent="outline"><span>This is a content slot</span></gux-card-beta>',
    '<gux-card-beta accent="raised"><span>This is a content slot</span></gux-card-beta>',
    '<gux-card-beta accent="filled"><span>This is a content slot</span></gux-card-beta>'
  ].forEach((html, index) => {
    it(`should render component as expected (${index + 1})`, async () => {
      const page = await newSparkE2EPage({ html });
      const element = await page.find('gux-card-beta');

      await a11yCheck(page);

      expect(element.outerHTML).toMatchSnapshot();
    });
  });
});
