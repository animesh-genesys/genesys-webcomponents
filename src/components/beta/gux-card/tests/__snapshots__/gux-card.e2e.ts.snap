// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render different card variations eg(outline,outline(explicit),filled,raised) should render component as expected (1) 1`] = `"<gux-card-beta hydrated=\\"\\"><span>This is a content slot</span></gux-card-beta>"`;

exports[`#render different card variations eg(outline,outline(explicit),filled,raised) should render component as expected (2) 1`] = `"<gux-card-beta accent=\\"outline\\" hydrated=\\"\\"><span>This is a content slot</span></gux-card-beta>"`;

exports[`#render different card variations eg(outline,outline(explicit),filled,raised) should render component as expected (3) 1`] = `"<gux-card-beta accent=\\"raised\\" hydrated=\\"\\"><span>This is a content slot</span></gux-card-beta>"`;

exports[`#render different card variations eg(outline,outline(explicit),filled,raised) should render component as expected (4) 1`] = `"<gux-card-beta accent=\\"filled\\" hydrated=\\"\\"><span>This is a content slot</span></gux-card-beta>"`;
